tests/renderer/app.test.js

```diff
@@ -339,87 +342,124 @@ describe('Display recently executed InVEST jobs on Home tab', () => {
     });
     const recentJobs = await InvestJob.saveJob(job1);
 
-    const { getByText, findByText, getByTitle } = render(<App />);
+    const { getByText, findByText, getByRole } = render(<App />);
 
     await waitFor(() => {
       recentJobs.forEach((job) => {
         expect(getByText(job.argsValues.workspace_dir))
           .toBeTruthy();
       });
     });
-    fireEvent.click(getByTitle('settings'));
+    fireEvent.click(getByRole('button', { name: 'settings' }));
     fireEvent.click(getByText('Clear'));
     const node = await findByText(/button to setup a model/);
     expect(node).toBeInTheDocument();
   });
 });
 
 describe('InVEST global settings: dialog interactions', () => {
+  const nWorkersLabelText = 'Taskgraph n_workers parameter';
+  const loggingLabelText = 'Logging threshold';
+
   beforeEach(async () => {
     getInvestModelNames.mockResolvedValue({});
   });
+
   afterEach(async () => {
     jest.resetAllMocks();
   });
-  test('Invest settings: cancel, save, and invalid nWorkers', async () => {
+
+  test('Invest settings: change & Cancel', async () => {
     const nWorkers = '2';
     const loggingLevel = 'DEBUG';
-    const nWorkersLabelText = 'Taskgraph n_workers parameter';
-    const loggingLabelText = 'Logging threshold';
-    const badValue = 'a';
 
     const {
-      getByText, getByLabelText, getByTitle, findByTitle
+      getByText, getByLabelText, findByRole
     } = render(
       <App />
     );
 
-    fireEvent.click(await findByTitle('settings'));
+    userEvent.click(await findByRole('button', { name: 'settings' }));
```

Comment:
Out of curiosity, was there a particular reason you switched to `userEvent`? I wish the docs were more helpful in establishing the differences between the two.
>Most projects have a few use cases for fireEvent, but the majority of the time you should probably use @testing-library/user-event.

Reply:
The difference as I understand it (don't remember where I read it) is fireEvent fires the single event (e.g. "click") and userEvent mimics the actual user interactions (e.g. mousedown, mouseup) which result in the click event.

Basically this test was giving me issues for other reasons and it was kind of big and unwieldy, so I split it up a bit and made this update in the process.